common/src/test/java/org/opensearch/ml/common/transport/upload/MLUploadInputTest.java

```diff
@@ -0,0 +1,178 @@
+        MLUploadInput.builder()
+                .functionName(functionName)
+                .modelName(null)
+                .build();
+    }
+
+    @Test
+    public void constructor_NullModelVersion() {
+        exceptionRule.expect(IllegalArgumentException.class);
+        exceptionRule.expectMessage("model version is null");
+        MLUploadInput.builder()
+                .functionName(functionName)
+                .modelName(modelName)
+                .version(null)
+                .build();
+    }
+
+    @Test
+    public void constructor_NullModelFormat() {
+        exceptionRule.expect(IllegalArgumentException.class);
+        exceptionRule.expectMessage("model format is null");
+        MLUploadInput.builder()
+                .functionName(functionName)
+                .modelName(modelName)
+                .version(version)
+                .modelFormat(null)
+                .build();
+    }
+
+    @Test
+    public void constructor_NullModelConfig() {
+        exceptionRule.expect(IllegalArgumentException.class);
+        exceptionRule.expectMessage("model config is null");
+        MLUploadInput.builder()
+                .functionName(functionName)
+                .modelName(modelName)
+                .version(version)
+                .modelFormat(MLModelFormat.ONNX)
+                .modelConfig(null)
+                .build();
+    }
+
+    @Test
+    public void constructor_NullModelFileUrl() {
+        exceptionRule.expect(IllegalArgumentException.class);
+        exceptionRule.expectMessage("model file url is null");
+        MLUploadInput.builder()
+                .functionName(functionName)
+                .modelName(modelName)
+                .version(version)
+                .modelFormat(MLModelFormat.ONNX)
+                .modelConfig(config)
+                .url(null)
+                .build();
+    }
+
+    @Test
+    public void constructor_SuccessWithMinimalSetup() {
+        MLUploadInput input = MLUploadInput.builder()
+                .modelName(modelName)
+                .version(version)
+                .modelFormat(MLModelFormat.ONNX)
+                .modelConfig(config)
+                .url(url)
+                .build();
+        // MLUploadInput.functionName is set to FunctionName.TEXT_EMBEDDING if not explicitly passed, with no exception thrown
+        assertEquals(FunctionName.TEXT_EMBEDDING, input.getFunctionName());
+        // MLUploadInput.loadModel is set to false if not explicitly passed, with no exception thrown
+        assertFalse(input.isLoadModel());
+        // MLUploadInput.loadModel is set to null if not explicitly passed, with no exception thrown
+        assertNull(input.getModelNodeIds());
+    }
+
+/*
+
+    @Test
+    public void testParse1() throws Exception {
+        XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON);
+        input.toXContent(builder, ToXContent.EMPTY_PARAMS);
+    }
```

Comment:
This test fail because MLModelConfig provided is a mocked instance, which is null. This null value will trigger an error when you want to parse a field. If you provide a real instance for MLModelConfig in the MLUploadInput input, the test will pass.